File: backend/app/models/activity_log.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, JSON, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base

class ActivityLog(Base):
    __tablename__ = "activity_logs"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    organization_id = Column(Integer, ForeignKey("organizations.id"), nullable=False)
    
    action = Column(String(50), nullable=False)  # create, update, delete, generate, export, login
    entity_type = Column(String(50))  # brand, project, post, user
    entity_id = Column(Integer)
    entity_name = Column(String(255))
    
    details = Column(JSON)  # Dettagli aggiuntivi
    ip_address = Column(String(45))
    user_agent = Column(Text)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    user = relationship("User")

File: app/Domain/Organization/Models/ActivityLog.php
<?php

declare(strict_types=1);

namespace App\Domain\Organization\Models;

use App\Domain\User\Models\User;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ActivityLog extends Model
{
    const UPDATED_AT = null;

    protected $fillable = [
        'user_id',
        'organization_id',
        'action',
        'entity_type',
        'entity_id',
        'entity_name',
        'details',
        'ip_address',
        'user_agent',
    ];

    protected function casts(): array
    {
        return [
            'details'    => 'array',
            'created_at' => 'datetime',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function organization(): BelongsTo
    {
        return $this->belongsTo(Organization::class);
    }

    public static function log(
        int $userId,
        int $organizationId,
        string $action,
        ?string $entityType = null,
        ?int $entityId = null,
        ?string $entityName = null,
        array $details = [],
        ?string $ip = null,
        ?string $userAgent = null,
    ): self {
        return self::create([
            'user_id'         => $userId,
            'organization_id' => $organizationId,
            'action'          => $action,
            'entity_type'     => $entityType,
            'entity_id'       => $entityId,
            'entity_name'     => $entityName,
            'details'         => $details ?: null,
            'ip_address'      => $ip,
            'user_agent'      => $userAgent,
        ]);
    }
}
